sdk/core/azure-core/src/main/java/com/azure/core/util/TransferUtil.java

```diff
@@ -0,0 +1,110 @@
+     * @return A {@link Mono} which completion indicates successful transfer.
+     */
+    public static Mono<Void> downloadToAsynchronousByteChannel(
+        AsynchronousByteChannel channel,
+        Mono<StreamResponse> responseMono,
+        BiFunction<Exception, Long, Mono<StreamResponse>> onDownloadErrorResume,
+        ProgressReporter progressReporter,
+        int maxRetries) {
+
+        return downloadToAsynchronousByteChannel(
+            new ByteCountingAsynchronousByteChannel(channel, progressReporter), responseMono, onDownloadErrorResume, maxRetries, 0);
+    }
+
+    private static Mono<Void> downloadToAsynchronousByteChannel(
+        ByteCountingAsynchronousByteChannel channel,
+        Mono<StreamResponse> responseMono,
+        BiFunction<Exception, Long, Mono<StreamResponse>> onDownloadErrorResume,
+        int maxRetries, int retryCount) {
+
+        return responseMono
+            .flatMap(response -> response.transferContentToAsync(channel))
+            .onErrorResume(Exception.class, exception -> {
+                int updatedRetryCount = retryCount + 1;
+
+                if (updatedRetryCount > maxRetries) {
+                    return Mono.error(exception);
+                }
+
+                return downloadToAsynchronousByteChannel(
+                    channel, onDownloadErrorResume.apply(exception, channel.getBytesWritten()),
+                    onDownloadErrorResume, maxRetries, updatedRetryCount);
+            });
+    }
+
+    /**
+     * Transfers the {@link StreamResponse} content to {@link WritableByteChannel}.
+     * Resumes the transfer in case of errors.
+     *
+     * @param channel The destination {@link WritableByteChannel}.
+     * @param response The initial response.
+     * @param onDownloadErrorResume A {@link BiFunction} of {@link Throwable} and {@link Long} which is used to resume
+     *                              downloading when an error occurs.
+     * @param progressReporter The {@link ProgressReporter}.
+     * @param maxRetries The maximum number of times a download can be resumed when an error occurs.
+     */
+    public static void downloadToWritableByteChannel(
+        WritableByteChannel channel,
+        StreamResponse response,
+        BiFunction<Exception, Long, StreamResponse> onDownloadErrorResume,
+        ProgressReporter progressReporter,
+        int maxRetries) {
+
+        downloadToWritableByteChannel(
+            new ByteCountingWritableByteChannel(channel, progressReporter), response, onDownloadErrorResume, maxRetries, 0);
+    }
+
+    private static void downloadToWritableByteChannel(
+        ByteCountingWritableByteChannel channel,
+        StreamResponse response,
+        BiFunction<Exception, Long, StreamResponse> onDownloadErrorResume,
+        int maxRetries, int retryCount) {
+
+        try {
+            response.transferContentTo(channel);
+        } catch (RuntimeException e) {
```

Comment:
This is different than the Async case, async is retrying on `Exception`

Reply:
`transferContentTo` doesn't declare any checked exceptions here.